llama_index/synthesizers/base.py

```diff
@@ -0,0 +1,186 @@
+"""Response builder class.
+
+This class provides general functions for taking in a set of text
+and generating a response.
+
+Will support different modes, from 1) stuffing chunks into prompt,
+2) create and refine separately over each chunk, 3) tree summarization.
+
+"""
+import logging
+from abc import ABC, abstractmethod
+from typing import Any, Dict, Generator, List, Optional, Sequence, Union
+
+from llama_index.indices.service_context import ServiceContext
+from llama_index.callbacks.schema import CBEventType, EventPayload
+from llama_index.indices.postprocessor.types import BaseNodePostprocessor
+from llama_index.indices.query.schema import QueryBundle
+from llama_index.indices.service_context import ServiceContext
+from llama_index.response.schema import RESPONSE_TYPE, Response, StreamingResponse
+from llama_index.schema import BaseNode, NodeWithScore, MetadataMode
+from llama_index.token_counter.token_counter import llm_token_counter
+from llama_index.types import RESPONSE_TEXT_TYPE
+
+logger = logging.getLogger(__name__)
+
+QueryTextType = Union[str, QueryBundle]
+
+
+class BaseSynthesizer(ABC):
+    """Response builder class."""
+
+    def __init__(
+        self,
+        service_context: Optional[ServiceContext] = None,
+        node_postprocessors: Optional[List[BaseNodePostprocessor]] = None,
```

Comment:
should node_postprocessors be here at all? tbh i'd be in favor of stripping it out entirely, having it as its own arg in `RetrieverQueryEngine`

Reply:
Hmm, yea i think I like that! Originally I was thinking each response synthesizer would be responsible for running it, but maybe it's better to decouple that a bit more

Reply:
yeah i think this makes sense to decouple retrieval from synthesis a bit more